pkg/agent/openflow/pipeline.go

```diff
@@ -181,6 +180,23 @@ func GetTableList() []binding.Table {
 	return tables
 }
 
+// CleanOFTableCache is used to reset ofTableCache and only used in integration tests. When all integration tests about
+// openflow run in batch, unexpected flows could be installed on OVS due to stale ofTableCache, which may cause some tests
+// to fail. For example, for TestFuncA, SNATTable is needed; for TestFuncB, SNATTable is not needed. If TestFuncB is run
+// after TestFuncA, since ofTableCache (SNATTable is added by TestFuncA) is not reset, default flow of SNATTable will also
+// be realized on OVS when running TestFuncB (see "func (c *client) defaultFlows() (flows []binding.Flow)"). Note that,
+// the unexpected flows are not included in the map tableCache of OFBridge defined in pkg/ovs/openflow/ofctrl_bridge.go,
+// because the bridge will be destroyed after every test. For some tests, function checkOVSFlowMetrics (defined in
+// test/integration/agent/openflow_test.go) is used to check the flow number of every installed table. The expected table
+// list is read from the map tableCache of OFBridge, but the actual table list is dumped from OVS bridge (including the
+// unexpected flow). They are different, and as a result, TestFuncB will fail.
+func CleanOFTableCache() {
```

Comment:
I don't think this function `CleanOFTableCache` is relevant with unused OF tables/flows on Windows, and it looks an improvement on the existing integration test code. If yes, better to mention this improvement in the commitment.

Reply:
Yes, it's not relevant with unused OF tables/flows on Windows. I'll update the commitment.

Reply:
This is exactly one of the reasons why global variables for status is strongly discouraged. It makes testing really difficult. With such global variables, more and more reset/cleanup utils have to be introduced to hack it. A well designed object should encapsulate its state internally and all state should be auto-destroyed when the struct is no longer used.

Reply:
Maybe we need a PR to remove such global variables in the future.